core/src/components/breadcrumb/breadcrumb.tsx

```diff
@@ -0,0 +1,212 @@
+   */
+  @Event() ionFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the breadcrumb loses focus.
+   */
+  @Event() ionBlur!: EventEmitter<void>;
+
+  /**
+   * Emitted when the collapsed indicator is clicked on.
+   */
+  @Event() ionCollapsedClick!: EventEmitter<void>;
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAttributes(this.el, ['aria-label']);
+  }
+
+  private onFocus = () => {
+    this.ionFocus.emit();
+  }
+
+  private onBlur = () => {
+    this.ionBlur.emit();
+  }
+
+  private collapsedIndicatorClick = () => {
+    this.ionCollapsedClick.emit();
+  }
+
+  render() {
+    const { color, active, collapsed, disabled, download, el, inheritedAttributes, last, routerAnimation, routerDirection, separator, showCollapsedIndicator, target } = this;
+    const TagType = this.href === undefined ? 'span' : 'a' as any;
+
+    // Links can still be tabbed to when set to disabled if they have an href
+    // in order to truly disable them we can keep it as an anchor but remove the href
+    const href = disabled ? undefined : this.href;
+    const mode = getIonMode(this);
+    const attrs = (TagType === 'span')
+      ? { }
+      : {
+        download,
+        href,
+        target
+      };
+
+    // If the breadcrumb is collapsed, check if it contains the collapsed indicator
+    // to show the separator as long as it isn't also the last breadcrumb
+    // otherwise if not collapsed use the value in separator
+    const showSeparator = last
+      ? false
+      : collapsed
+        ? showCollapsedIndicator && !last ? true : false
+        : separator;
+
+    return (
+      <Host
+        onClick={(ev: Event) => openURL(href, ev, routerDirection, routerAnimation)}
+        aria-disabled={disabled ? 'true' : null}
+        class={createColorClasses(color, {
+          [mode]: true,
+          'breadcrumb-active': active,
+          'breadcrumb-collapsed': collapsed,
+          'breadcrumb-disabled': disabled,
+          'in-breadcrumbs-color': hostContext('ion-breadcrumbs[color]', el),
+          'in-toolbar': hostContext('ion-toolbar', this.el),
+          'in-toolbar-color': hostContext('ion-toolbar[color]', this.el),
+          'ion-activatable': true,
+          'ion-focusable': true,
+        })}
+      >
+        <TagType
+          {...attrs}
+          class="breadcrumb-native"
+          part="native"
+          disabled={disabled}
+          onFocus={this.onFocus}
+          onBlur={this.onBlur}
+          {...inheritedAttributes}
+        >
+          <slot name="start"></slot>
+          <slot></slot>
+          <slot name="end"></slot>
+        </TagType>
+        { showCollapsedIndicator &&
+            <button
+              part="collapsed-indicator"
+              onClick={() => this.collapsedIndicatorClick()}
+              class={{
+                'breadcrumbs-collapsed-indicator': true,
+              }}
+            >
+              <ion-icon name="ellipsis-horizontal" lazy={false}></ion-icon>
```

Comment:
We should opt to use the ES6 import instead so that icons work properly with React and Vue. In other components we provided the string name here and then manually bundled the icons used in the main chunk of Ionic React and Vue, but if we do the ES6 import here it _should_ allow us to not have to manually bundle the icons and allow treeshaking to work better.

```suggestion
              <ion-icon icon={ellipsisHorizontal} lazy={false}></ion-icon>
```
`ellipsisHorizontal` is imported from `ionicons/icons`.